src/riot-rs-threads/src/smp/rp2040.rs

```diff
@@ -0,0 +1,98 @@
+use crate::arch::{Arch as _, Cpu};
+
+use cortex_m::peripheral::SCB;
+use embassy_rp::{
+    interrupt,
+    interrupt::InterruptExt as _,
+    multicore::{spawn_core1, Stack},
+    peripherals::CORE1,
+};
+use rp_pac::SIO;
+use static_cell::ConstStaticCell;
+
+use super::{CoreId, Multicore};
+
+pub struct Chip;
+
+impl Multicore for Chip {
+    const CORES: u32 = 2;
+
+    fn core_id() -> CoreId {
+        CoreId(SIO.cpuid().read() as u8)
+    }
+
+    fn startup_other_cores() {
+        // TODO: How much stack do we really need here?
+        static STACK: ConstStaticCell<Stack<4096>> = ConstStaticCell::new(Stack::new());
```

Comment:
Out-of-band discussion concerning stack size: 
- If the `interrupt-executor` is used, it runs only on core0 and all embassy interrupts are handled there. Thus only little stack is needed for the second core in that case. (@kaspar030 how much do you think is reasonable?)
- In case of the thread executor the needed stack very much depends on where the embassy init thread runs, which other interrupts are configured on the second core, and how much stack their handler requires. There is a general plan to in the future add logic for picking reasonable stack sizes, but that's out of scope for this PR and also independent of multicore. For now, let's pick the same `ISR_STACKSIZE` as on core 0? 

And maybe add `ISR_STACKSIZE_CORE1` env so it can be overridden by a user?

Reply:
Added env-based `ISR_STACKSIZE_CORE1` constant, that fallbacks to  `ISR_STACKSIZE`.